server/app/services/ambassadors.js

```diff
@@ -303,46 +302,70 @@ async function sendTriplerCountsToHubspot(ambassador) {
  */
 
 async function updateEkataMatchScore(ambassador) {
-  let ambassadorSubScore = 0
-  let triplerSubScore = 0
-  let ekataMatchScore = 0
-  let ekataThreshold = 3
-  let ekataPenalty = 0
+  console.log("in updateEkataMatchScore")
+  //resetting (in case any validation score changed, it calculates fresh)
+  let ambassadorMatches = 0
+  let tripler_ekata_blemish = 0
+  let ambassador_ekata_blemish = 0
+  let re = /\d* \w{1,2}/
+
+  //these are still being tuned to find the best fit
+
+  let ambassadorEkataThreshold = ov_config.ambassadorEkataThreshold
+    ? ov_config.ambassadorEkataThreshold
+    : 1
+  let triplerEkataPenalty = ov_config.triplerEkataPenalty ? ov_config.triplerEkataPenalty : 1
+  let ambassadorEkataPenalty = ov_config.ambassadorEkataPenalty
+    ? ov_config.ambassadorEkataPenalty
+    : 2
+  let triplerEkataBonus = ov_config.triplerEkataBonus ? ov_config.triplerEkataBonus : -1
 
-  // calculate Ambassador's Tripler SubScore
   let query =
-    "MATCH (a:Ambassador {id:$a_id})-[:CLAIMS]->(t:Tripler) RETURN t.first_name as first_name, t.last_name as last_name, t.address as address, t.verification as verification"
+    "MATCH (a:Ambassador {id:$a_id})-[:CLAIMS]->(t:Tripler) WHERE  t.status <> 'unconfirmed' RETURN t.first_name as first_name, t.last_name as last_name, t.address as address, t.verification as verification"
   let collection = await neode.cypher(query, {a_id: ambassador.get("id")})
   let verificationStrings = []
   for (let index = 0; index < collection.records.length; index++) {
+    let invidualTriplerMatchScore = 0
+    let address = collection.records[index].get("address").toLowerCase()
     let verificationString = collection.records[index].get("verification").toString().toLowerCase()
     let triplerProperties = []
     triplerProperties.push(collection.records[index].get("first_name").toLowerCase())
     triplerProperties.push(collection.records[index].get("last_name").toLowerCase())
-
+    triplerProperties.push(address.match(re)[0])
     for (let i in triplerProperties) {
-      if (verificationString.includes(verificationString[i])) {
-        triplerSubScore += 1
-      } else {
-        triplerSubScore += -1
+      if (verificationString.includes(triplerProperties[i])) {
+        invidualTriplerMatchScore = ++invidualTriplerMatchScore
       }
     }
+
+    if (invidualTriplerMatchScore >= 1) {
```

Comment:
Why is this incrementing `tripler_ekata_blemish` rather than just setting it? Did I miss a change above or is this just here in case we decide to do something different with it above?

Reply:
so that it tripler_ekata_blemish on the ambassador increases / decreases as individual triplers are processed

Reply:
But it's set to zero at 308 and then incremented here. I'm not saying it's wrong, it was just confusing to me. Looks like mathematically though.

Reply:
this section is within the for loop that's iterating through the individual triplers. 
by the time the second tripler comes, tripler_ekata_blemish will have been updated with either a penalty or a bonus, and we would need to update it with the new penalty or bonus from the second tripler

Reply:
Right. I messed up the loop location.